Userland/Services/SpiceAgent/VDAgent.h

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2021, Kyle Pereira <kyle@xylepereira.me>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "ClipboardServerConnection.h"
+#include <AK/ByteBuffer.h>
+#include <AK/Vector.h>
+#include <LibCore/Notifier.h>
+
+#pragma once
+
+// Fun fact, I have no idea what VD stands for.
+class VDAgent {
+public:
+    VDAgent(int fd, ClipboardServerConnection&);
+
+    static constexpr u32 AGENT_PROTOCOL = 1;
+    enum class Port {
+        Client = 1,
+        Server
+    };
+
+    struct [[gnu::packed]] ChunkHeader {
+        u32 port {};
+        u32 size {};
+    };
+
+    struct [[gnu::packed]] Message {
+        u32 protocol {};
+        u32 type {};
+        u64 opaque {};
+        u32 size {};
```

Comment:
theres no need to explicitly initialize these primitives, as these are always read in from existing data.